common/singleChunkReader.go

```diff
@@ -227,14 +227,20 @@ func (cr *singleChunkReader) blockingPrefetch(fileReader io.ReaderAt, isRetry bo
 	// Must use "relaxed" RAM limit IFF this is a retry.  Else, we can, in theory, get deadlock with all active goroutines blocked
 	// here doing retries, but no RAM _will_ become available because its
 	// all used by queued chunkfuncs (that can't be processed because all goroutines are active).
-	cr.chunkLogger.LogChunkStatus(cr.chunkId, EWaitReason.RAMToSchedule())
+	if cr.chunkLogger != nil {
+		cr.chunkLogger.LogChunkStatus(cr.chunkId, EWaitReason.RAMToSchedule())
+	}
+
 	err := cr.cacheLimiter.WaitUntilAdd(cr.ctx, cr.length, func() bool { return isRetry })
```

Comment:
Just Curious: Can cr.chunkLogger be nil? If yes, can cr.cacheLimiter also be nil? If yes, Why not have nil check for this line cr.cacheLimiter?

Reply:
We are setting cr.chunkLogger to nil, once it done the transfer. So that it release the reference of chunkStatusLogger. 
cr.cacheLimiter never set to nil, so no need to check that.